src/afl-fuzz-state.c

```diff
@@ -507,8 +507,8 @@ void afl_states_stop(void) {
 
   LIST_FOREACH(&afl_states, afl_state_t, {
 
-    if (el->fsrv.child_pid > 0) kill(el->fsrv.child_pid, SIGKILL);
-    if (el->fsrv.fsrv_pid > 0) kill(el->fsrv.fsrv_pid, SIGKILL);
+    if (el->fsrv.child_pid > 0) kill(el->fsrv.child_pid, el->fsrv.kill_signal);
```

Comment:
I think it needs a second variable `AFL_DO_NOT_KILL_CHILD` or something like that? I think I remember @tklengyel writing that the child may not be killed for Xen mode.

Reply:
This was resolved on xen mode side

Reply:
Yes, this is not an issue anymore, we report the forkserver's pid as the child's pid and that just works. The forkserver receives the signal twice here but that doesn't matter.